nodes/core/io/21-httpin.js

```diff
@@ -232,6 +250,15 @@ module.exports = function(RED) {
                     msg.payload += chunk;
                 });
                 res.on('end',function() {
+                    if(RED.settings.metricsOn && RED.settings.metricsOn === true) {
+                        if(preRequestTimestamp) {
+                            var elapsedTimeDiff = process.hrtime(preRequestTimestamp);
+                            node.metric(msg, "request.duration.nanos", (elapsedTimeDiff[0] * 1e9 + elapsedTimeDiff[1]));
```

Comment:
Note I will also be changing the order of the arguments:

    node.metric(eventName, msg, metrics)

The name really should be the first field - looks wrong otherwise.